### PR TITLE
Chore: add addmv_ to tensor 

### PR DESCRIPTION
**Pull request by a0m0rajab**
Closes https://github.com/unifyai/ivy/issues/20959

@hirwa-nshuti closed the PR since I broke the Git history.

Previous review: https://github.com/unifyai/ivy/pull/20998

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -614,6 +614,59 @@ def test_torch_addmv(
...
+)
+def test_torch_addmv_(
+    dtype_and_matrices,
+    beta,
+    alpha,
+    frontend,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    on_device,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:646]:
```suggestion
    on_device,
    backend_fw,
```
There is a missing fixture for backend to test

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -614,6 +614,59 @@ def test_torch_addmv(
...
+):
+    input_dtype, x, mat, vec = dtype_and_matrices
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        init_all_as_kwargs_np={
+            "data": x,
+        },
+        method_input_dtypes=input_dtype,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:654]:
```suggestion
        method_input_dtypes=input_dtype,
        backend_to_test=backend_fw,
```
This argument was missing

File: ivy/functional/frontends/torch/tensor.py [hunk 3]
```diff
@@ -173,6 +173,11 @@ def addmm_(self, mat1, mat2, *, beta=1, alpha=1):
     def addmv(self, mat, vec, *, beta=1, alpha=1):
         return torch_frontend.addmv(self, mat, vec, beta=beta, alpha=alpha)
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+    def addmv_(self, mat, vec, *, beta=1, alpha=1):
+        self.ivy_array = torch_frontend.addmv(self, mat, vec, beta=beta, alpha=alpha)
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/torch/tensor.py:178]: Here we need to ensure that we are assigning `self.ivy_array` to correct `ivy.Array` instance
```suggestion
        self.ivy_array = torch_frontend.addmv(self, mat, vec, beta=beta, alpha=alpha).ivy_array
```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @a0m0rajab 

Thanks for making the requested changes😊
I added a few suggestions after they're added will get the PR merged.
Thanks🙂

**Comment by a0m0rajab** [comment]
@hirwa-nshuti, thanks for that, I just applied the suggestions.

**Comment by a0m0rajab** [comment]
Here is the failing test logs, @hirwa-nshuti what's your input on this?
```logs
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_addmv_,numpy
****************************************************************************************************

ImportError while loading conftest '/ivy/ivy_tests/test_ivy/conftest.py'.
ivy_tests/test_ivy/conftest.py:7: in <module>
    import ivy_tests.test_ivy.helpers.test_parameter_flags as pf
ivy_tests/test_ivy/helpers/__init__.py:2: in <module>
    from . import hypothesis_helpers
ivy_tests/test_ivy/helpers/hypothesis_helpers/__init__.py:1: in <module>
    from . import general_helpers
ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:8: in <module>
    import ivy
ivy/__init__.py:770: in <module>
    from . import stateful
ivy/stateful/__init__.py:1: in <module>
    from . import activations
ivy/stateful/activations.py:5: in <module>
    from ivy.stateful.module import Module
ivy/stateful/module.py:8: in <module>
    import dill
E   ModuleNotFoundError: No module named 'dill'
Error: Process completed with exit code 1.
```

**Comment by a0m0rajab** [comment]
I did not use such `dill` module in my PR.

**Comment by hirwa-nshuti** [comment]
> Here is the failing test logs, @hirwa-nshuti what's your input on this?
> 
> ```
> ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_addmv_,numpy
> ****************************************************************************************************
> 
> ImportError while loading conftest '/ivy/ivy_tests/test_ivy/conftest.py'.
> ivy_tests/test_ivy/conftest.py:7: in <module>
>     import ivy_tests.test_ivy.helpers.test_parameter_flags as pf
> ivy_tests/test_ivy/helpers/__init__.py:2: in <module>
>     from . import hypothesis_helpers
> ivy_tests/test_ivy/helpers/hypothesis_helpers/__init__.py:1: in <module>
>     from . import general_helpers
> ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:8: in <module>
>     import ivy
> ivy/__init__.py:770: in <module>
>     from . import stateful
> ivy/stateful/__init__.py:1: in <module>
>     from . import activations
> ivy/stateful/activations.py:5: in <module>
>     from ivy.stateful.module import Module
> ivy/stateful/module.py:8: in <module>
>     import dill
> E   ModuleNotFoundError: No module named 'dill'
> Error: Process completed with exit code 1.
> ```

Try to rebuild your container as this dependency was added recently

**Review by hirwa-nshuti — APPROVED** [review]
Hey @a0m0rajab 

All looks great now 🙂
Going to merge😊

**Comment by a0m0rajab** [comment]
Thank you for the merge!

**Pull request merged**